### DISCUSSION TITLE
JSONB support

### DISCUSSION THREAD

**Discussion by ngortheone** [zombiezen/go-sqlite#82] [Q&A] [answered]
sqlite has recently landed JSONB support.
https://sqlite.org/forum/forumpost/fa6f64e3dc1a5d97

Is this feature available in the go implementation?

**Comment by zombiezen** [comment, marked as answer]
Yup! As of [v1.1.2](https://github.com/zombiezen/go-sqlite/releases/tag/v1.1.2), we pull in `modernc.org/sqlite` 1.29.0, which is equivalent to upstream SQLite [3.45.1](https://sqlite.org/releaselog/3_45_1.html).

**Reply by ngortheone** [reply]
Is there an example of go API how to work with jsonb?

**Reply by zombiezen** [reply]
There's not a Go (or C)-specific API for JSONB because AFAIK everything is done through SQL functions: https://sqlite.org/json1.html

**Reply by ngortheone** [reply]
Yeah, I am being stupid. For others like me what I was looking for is this:

```sql
CREATE TABLE users (
  id INTEGER PRIMARY KEY,
  data BLOB
);
```

```sql
INSERT INTO users (id, data) VALUES (1, jsonb('{
  "name": "Alice",
  "age": 25,
  "email": "alice@example.com"
}'));
```

and then 

```
select id, data -> 'age' from users;
```

or 

```sql
select id, json(data) from users;
```

**Reply by ngortheone** [reply]
can be tried here: https://sqliteonline.com/
